snapshot.go

```diff
@@ -315,11 +220,20 @@ func getAttemptDownloadClosure(isBoot bool, snapshot *common.Snapshot, uri strin
 
 func processSnapshotServerFileResponse(dbId string, body io.Reader, snapshot *common.Snapshot) error {
 	dbPath := data.DBPath("common/" + dbId)
+	dbDir := dbPath[0 : len(dbPath)-7]
 	log.Infof("Attempting to stream the sqlite snapshot to %s", dbPath)
 
+	// if exists, delete the old snapshot file
+	if _, err := os.Stat(dbDir); !os.IsNotExist(err) {
+		if err = os.RemoveAll(dbDir); err != nil {
```

Comment:
Are you certain old copy is all processed and no fallback needed?

Reply:
Good catch!
 In fact, it should only be removed for the bootstrap snapshot. For data snapshot, in theory we will never receive duplicate snapshot#. However, we did observe a bug before when proxy heartbeat failed the Snapshot Server may give duplicate snapshot#.
That's really a good catch. . I'll updated it with `if isBoot`, just in case similar bugs in SS crush apid.